### PR TITLE
feat(trading): merge deal ticket stores

### LINKED ISSUE #4451: Merge deal ticket stores
## The Chore

Stores that are storing data connected to deal ticket form should be merged into one deal ticket store and removed from orders lib.

### PR DESCRIPTION
**Pull request by bglownia**
# Related issues 🔗

Closes #4451

# Technical 👨‍🔧

Merge three zustand stores into one. Change flow of data in deal ticket, use store as side effect.

### CODE REVIEW

**Comment by mattrussell36** [comment]
As discussed, it would be good if the hierarchy within the store state should be markets -> type -> fields

I think that would be cleaner in the code but also give us separated persistance between the different order types in the deal ticket

File: libs/deal-ticket/src/components/deal-ticket/deal-ticket-stop-order.tsx [hunk 1]
```diff
@@ -50,32 +48,40 @@ export interface StopOrderProps {
...
-
 export const StopOrder = ({ market, marketPrice, submit }: StopOrderProps) => {
   const { pubKey, isReadOnly } = useVegaWallet();
-  const setDealTicketType = useDealTicketTypeStore((state) => state.set);
-  const [, updateOrder] = useOrder(market.id);
-  const updateStoredFormValues = useStopOrderFormValues(
-    (state) => state.update
+  const setType = useDealTicketFormValues((state) => state.setType);
+  const dealTicketType = useDealTicketFormValues((state) =>
+    state.formValues[market.id]?.type === DealTicketType.StopMarket
```
Comment (mattrussell36) [inline libs/deal-ticket/src/components/deal-ticket/deal-ticket-stop-order.tsx:67]: What if the stored ticket type is normal limit or normal market? This doesn't look right to me?

File: libs/deal-ticket/src/hooks/use-form-values.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+          set((state) => {
+            state.formValues[marketId] = Object.assign(
+              state.formValues[marketId] ?? {},
+              { type }
+            );
+          });
+        },
+      })),
+      {
+        name: 'deal-ticket-form-values',
```
Comment (mattrussell36) [inline libs/deal-ticket/src/hooks/use-form-values.ts:137]: vega_deal_ticket_store for consistency

File: libs/deal-ticket/src/hooks/use-form-values.ts [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+export const isStopOrderType = (dealTicketType?: DealTicketType) =>
+  dealTicketType === DealTicketType.StopLimit ||
+  dealTicketType === DealTicketType.StopMarket;
+
+export const useDealTicketFormValues = create<Store>()(
+  immer(
+    persist(
+      subscribeWithSelector((set) => ({
+        formValues: {},
+        updateStopOrder: (marketId, formValues) => {
```
Comment (mattrussell36) [inline libs/deal-ticket/src/hooks/use-form-values.ts:87]: Its super unclear when to use updateStopOrder vs update vs updateOrder

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: libs/deal-ticket/src/hooks/use-form-values.ts (hunk 3 above)
Comment (bglownia) [reply libs/deal-ticket/src/hooks/use-form-values.ts:90]: I renamed `update` to `updateAll`.

**Review by mattrussell36 — APPROVED** [review]

**Pull request merged**